.github/actions/changelog/index.js

```diff
@@ -32,107 +33,131 @@ const get_author = (commit, authors) => {
 };
 
 try {
+  const token = core.getInput("token");
   const head_version = fs
     .readFileSync(core.getInput("version-path"), "utf8")
     .trim();
-  const commits = JSON.parse(core.getInput("commits-json"));
   const changelog = fs.readFileSync(core.getInput("changelog-path"), "utf8");
-  let curline = 0;
-  const begin = 8;
-  let end = Number.MAX_SAFE_INTEGER;
-  const rl = readline.createInterface({
-    input: fs.createReadStream(core.getInput("changelog-path")),
-  });
-  let patch = false;
-  let minor = false;
-  let noupdate = false;
-  rl.on("line", (text) => {
-    if (curline < begin || curline > end) {
-      curline++;
-      return;
-    }
-    if (text.match("\\b" + head_version + "\\b")) {
-      end = curline++;
-      rl.close();
-      return;
-    }
-    console.log(text);
-    if (
-      text.includes("### Added") ||
-      text.includes("### Changed") ||
-      text.includes("### Removed")
-    ) {
-      minor = true;
-      console.log("found minor version update");
-    }
-    if (text.includes("### Fixed")) {
-      patch = true;
-      console.log("found patch version update");
-    }
-    if (
-      text.includes("### Tests") ||
-      text.includes("### Docs") ||
-      text.includes("### Operations")
-    ) {
-      noupdate = true;
-      console.log("found noop header");
-    }
-    curline++;
-  });
-  rl.on("close", () => {
-    const semver = head_version.split(".");
-    let vmajor = semver[0];
-    let vminor = semver[1];
-    let vpatch = semver[2];
-    if (minor) {
-      vminor++;
-      vpatch = 0;
-    } else if (patch) {
-      vpatch++;
-    } else if (noupdate) {
-      //do nothing
-    } else {
-      core.setFailed(
-        "Changelog does not contain enough information to update the version."
-      );
-    }
-    const version = vmajor + "." + vminor + "." + vpatch;
-    const changelog_header =
-      "## [" + version + "] - " + new Date().toISOString().split("T")[0];
-    let message = "noop";
-    if (minor || patch) {
-      message = "The new version will be " + version;
-      unreleased = "UNRELEASED";
-      let commit_authors = "";
-      for (i = 0; i < commits.length; i++) {
-        commit_authors += get_author(commits[i], commit_authors);
-      }
+  const basehead = core.getInput("basehead");
+  const owner = github.context.payload.repository.organization
+  const repo = github.context.payload.repository.name
+  const octokit = github.getOctokit(token);
+  octokit.rest.repos
+    .compareCommitsWithBasehead({
+      owner: owner,
+      repo: repo,
+      basehead: basehead,
+    })
+    .then((compareCommits) => {
+      const {
+        data: { commits: commits },
+      } = compareCommits;
```

Comment:
This is the only meaningful change to the JS. The rest of the diff is indentation and minor changes in variable names.